gh-112419: Document removal of `sys.meta_path`'s `find_module` fallback (#112421)
python/cpython: Doc/library/sys.rst

    .. versionchanged:: 3.4

        :term:`Module specs <module spec>` were introduced in Python 3.4, by
        :pep:`451`.

    .. versionchanged:: 3.12

>         Earlier versions of Python looked for a method called
>         :meth:`!find_module` as fallback if a :data:`meta_path` entry doesn't
>         have a :meth:`~importlib.abc.MetaPathFinder.find_spec` method.
>         In this version, this behavior has been removed and the method wouldn't
>         be called as a fallback.

Reviewer: How about this?

```suggestion
        Removed the fallback that looked for a :meth:`!find_module` method
        if a :data:`meta_path` entry didn't have a
        :meth:`~importlib.abc.MetaPathFinder.find_spec` method.
```

        This is still called as a fallback if a :data:`meta_path` entry doesn't
        have a :meth:`~importlib.abc.MetaPathFinder.find_spec` method.

    .. versionchanged:: 3.12

>         :meth:`!find_module` has been removed and wouldn't be called as a

Reviewer 2: I think we should do the reverse thing: explain what happens now and in `.. versionchanged:: 3.12` we can add a note about how it is used to be.

Author: The main part of `sys.meta_path` describes what the current behavior is. In my understanding, you mean to remove the `find_path` in `.. versionchanged:: 3.4` and move it to `.. versionchanged:: 3.12` to show how it used to be?
